packages/core/src/reducers/core.ts

```diff
@@ -340,20 +340,47 @@ export const errorsAt = (
   schema: JsonSchema,
   matchPath: (path: string) => boolean
 ) => (errors: ErrorObject[]): ErrorObject[] => {
+  // Get data paths of oneOf and anyOf errors to later determine whether an error occurred inside a subschema of oneOf or anyOf.
   const combinatorPaths = filter(
     errors,
     error => error.keyword === 'oneOf' || error.keyword === 'anyOf'
   ).map(error => error.dataPath);
 
   return filter(errors, error => {
+    // Filter errors that match any keyword that we don't want to show in the UI
+    if (filteredErrorKeywords.indexOf(error.keyword) !== -1) {
+      return false;
+    }
+
     let result = matchPath(error.dataPath);
-    if (combinatorPaths.findIndex(p => instancePath.startsWith(p)) !== -1) {
-      result = result && isEqual(error.parentSchema, schema);
+    // In anyOf and oneOf blocks with "primitive subschemas" (not defining an object),
+    // we want to make sure that errors are only shown for the correct subschema.
+    // Therefore, we compare the error's parent schema with the property's schema.
+    // This is necessary because in the primitive case the error's data path is the same for all subschemas:
+    // It directly points to the property defining the anyOf/oneOf.
+    // In contrast, this comparison must not be done for errors whose parent schema defines an object
+    // because the parent schema can never match the property schema (e.g. for 'required' checks).
+    const parentSchema: JsonSchema | undefined = error.parentSchema;
+    if (result && parentSchema?.type !== 'object'
```

Comment:
There is also `type='array'` which is not a primitive. Also a schema could only define `properties` and therefore implicitly be an `object`.

Reply:
Good catch. I changed this to check for this in a separate method to match object schemas. We still want this check for array schemas in case one/anyOf subschemas use arrays. We only want to match array-level validations (e.g. minItems) for the corresponding sub schema.